Assets/Scripts/Character/HealthSystem.cs

```diff
@@ -0,0 +1,52 @@
+using UnityEngine;
+
+public class HealthSystem : MonoBehaviour
+{
+    [Header("Health Settings")]
+    public float maxHealth = 100.0f;
+    public float currentHealth;
+
+    [Header("Damage Settings")]
+    public LayerMask enemyLayers;
+    public float damageOnCollision = 10.0f;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = maxHealth;
+    }
+
+    public void TakeDamage(float damage)
+    {
+        currentHealth -= damage;
+
+        if (currentHealth <= 0)
+        {
+            Die();
+        }
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        // Check if the collided object is in the enemyLayers
+        if (((1 << other.gameObject.layer) & enemyLayers) != 0)
+        {
+            TakeDamage(damageOnCollision);
+        }
+    }
+
+    private void Die()
+    {
+        // Implement any logic needed when the object dies, e.g., destroying the object
+        if (gameObject.CompareTag("TopDownCharacter"))
```

Comment:
The tag manager has a “Player” tag already, you don’t need to create a new one. A good rule of thumb is when you’re considering tags there’s better alternatives. They have their uses but are commonly used for prototyping. That being said if you absolutely had to use tags, CompareTag is the way to go as it compiles into fewer instructions than myTag == tag. See https://answers.unity.com/questions/200820/is-comparetag-better-than-gameobjecttag-performanc.html